Bump 32-bit sunxi edge kernel to 6.3.8

viraniac opened the pull request:
# Description

Bump 32-bit kernel to 6.3.8. I didn't bumped it for 64-bit sunxi devices as drv-rgb-add-ws2812.patch introduced in #5267 makes use of of_get_named_gpio_flags function which is removed in 6.3. As I don't have a Bigtreetech cb1, I am not comfortable modifying code to fix this patch. The kernel module added by drv-rgb-add-ws2812.patch is not enabled for 32-bit devices and hence it doesn't cause compilation issues on 32-bit sunxi kernel. 

Excluding drv-rgb-add-ws2812.patch, I have updated any patch that was broken, and have disabled patches where changes were present upstream. Also disabled same rockchip related patches that I disabled in #5326 

# How Has This Been Tested?

- [X] Created an image and booted on Nanopi duo2

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

igorpecovnik commented:
> As I don't have a Bigtreetech cb1, I am not comfortable modifying code to fix this patch

I have asked them to check.

AGM1968 commented:
Hi Gunjan, Thanks.

I extracted megous  `git checkout -b orange-pi-6.3 remotes/origin/orangepi-pi-6.3 ` tagged  `Tag: v6.3.5`

The following failed against 6.3.5, haven't had time to work through this.

```
1:	INFO: [128] ?????? ????????? patches.megous/8723cs-Enable-WoWLAN.patch														
2:	INFO: [160] ?????? ????????? patches.megous/ARM-dts-sunxi-a83t-Add-SCPI-protocol.patch										3:	INFO: [164] ?????? ????????? patches.megous/ARM-dts-sun8i-a83t-tbs-a711-Give-Linux-more-privileges-over-SCP.patch		4:	INFO: [187] ?????? ????????? patches.megous/Bluetooth-Add-new-quirk-for-broken-local-ext-features-max_page.patch			5:	INFO: [188] ?????? ????????? patches.megous/f2fs-fix-unaligned-field-offset-in-32-bits-platform.patch						6:	INFO: [189] ?????? ????????? patches.megous/Bluetooth-btrtl-add-support-for-the-RTL8723CS.patch								7:	INFO: [212] ?????? ????????? patches.megous/regulator-axp20x-Turn-N_VBUSEN-to-input-on-x-powers-sense-vbus-.patch			8:	INFO: [213] ?????? ????????? patches.megous/power-supply-axp20x-usb-power-Support-input-current-limit.patch
9:	INFO: [239] ?????? ????????? patches.megous/ARM-dts-sunxi-h3-h5-Add-hdmi-sound-card.patch
10:	INFO: [240] ?????? ????????? patches.megous/iio-st_sensors-Don-t-report-error-when-the-device-is-not-presen.patch
11:	INFO: [249] ?????? ????????? patches.megous/arm64-dts-allwinner-h6-Add-hdmi-sound-card.patch
12:	INFO: [250] ?????? ????????? patches.megous/arm64-dts-allwinner-h6-Enable-hdmi-sound-card-on-boards-with-hd.patch
13:	INFO: [260] ?????? ????????? patches.megous/clk-sunxi-ng-a64-Increase-PLL_AUDIO-base-frequency.patch
14:	INFO: [261] ?????? ????????? patches.megous/sound-soc-sun8i-codec-Add-support-for-digital-part-of-the-AC100.patch
15:	INFO: [265] ?????? ????????? patches.megous/Move-a-node-to-avoid-merge-conflict.patch
16:	INFO: [355] ?????? ????????? patches.megous/usb-dwc3-Add-support-for-snps-usb3-phy-reset-quirk.patch						17:	INFO: [389] ?????? ????????? patches.megous/Revert-f2fs-fix-unaligned-field-offset-in-32-bits-platform.patch
18:	INFO: [402] ?????? ????????? patches.megous/Revert-Bluetooth-btrtl-add-support-for-the-RTL8723CS.patch
19:	INFO: [403] ?????? ????????? patches.megous/Revert-Bluetooth-Add-new-quirk-for-broken-local-ext-features-ma.patch
20:	

```
Extracted Megous Series list https://pastebin.com/00qM0mAC

viraniac commented:
> The following failed against 6.3.5, haven't had time to work through this.

@AGM1968 Sorry, bit confused with the message. Are you facing issues with the the patches from this merge request?

AGM1968 commented:
> Are you facing issues with the the patches from this merge request?

More a point of reference for your interest

viraniac commented:
Resynced megous patches from https://github.com/megous/linux/tree/orange-pi-6.3. Also made 6.3.9 as the edge kernel for both 32-bit and 64-bit allwinner boards except Bigtreetech CB1 which is kept back at 6.2.16 kernel

viraniac commented:
@AGM1968 Thanks, your comment made me realise that the previous changes were not correct. Previously, I had modified megous patches from 6.2 to apply on 6.3 kernel. Now I have updated them properly with megous kernel's 6.3 branch

I think I understand what is causing some patches to not get applied and needing to be edited manually. I think it might be a limitation of how mk_format_patch works and the order that it lists the patches in the series.conf file. I think I can fix it. Will raise the merge request tomorrow.

The-going commented:
> I think it might be a limitation of how mk_format_patch works and the order that it lists the patches in the series.conf file.

The script is working correctly. The problem is how the orange-pi branch was merged.
These were 27 other commit branches in which were created at different times.
For this reason, their order is a little mixed up. But the `git --format-patch` command won't do any better.
And this first merge was done on top of the initial kernel version (6.3).
Next, this resulting branch is merged with the upstream core and new commits from the upstream are on top. On the contrary, we always apply patches on top of the upstream. Patches have a fundamentally different order of application.

All you can do is a little sorting directly in the series file and then sequentially applying fix conflicts.
Some patches will go away during this procedure as they will already be in the upstream.
After all the patches are applied, extract them again with the mk_format_patch script. It will be a clean line.

viraniac commented:
I wanted it to add Kernel 6.4 to this merge request as well but sadly that is not going to cut it atleast today. Megous patches from [orange-pi-6.4-20230619-0323](https://github.com/megous/linux/releases/tag/orange-pi-6.4-20230619-0323) applied cleanly and didn't needed any fixing. I just filtered them to remove rockchip patches (simple sed, no manual work required). I fixed the armbian patches that needed fixing and yet 6.4 just refused to boot on my board resulting into same condition as initial armbian 23.05 image. Simply stuck to Starting kernel.

So while I am fixing the 6.4 kernel, lets atleast get 6.3 in.

schwar3kat reviewed (commented):
Orange Pi R1 (H2+)
Builds successfully
Boots successfully.
WiFi works (iperf3 results are as expected).
Both ethernet NIC's work (iperf3 results are as expected).

I have no issues with this being merged if other board maintainers agree.

igorpecovnik reviewed (approved):
ok, lets move this in
